kube/src/config/mod.rs

```diff
@@ -71,40 +92,140 @@ pub struct ConfigOptions {
     pub user: Option<String>,
 }
 
-/// Returns a config which is used by clients within pods on kubernetes.
-///
-/// It will return an error if called from out of kubernetes cluster.
-pub fn incluster_config() -> Result<Configuration> {
-    let server = incluster_config::kube_server().ok_or_else(|| {
-        Error::KubeConfig(format!(
-            "Unable to load incluster config, {} and {} must be defined",
-            incluster_config::SERVICE_HOSTENV,
-            incluster_config::SERVICE_PORTENV
-        ))
-    })?;
-
-    let default_ns = incluster_config::load_default_ns()
-        .map_err(|e| Error::KubeConfig(format!("Unable to load incluster default namespace: {}", e)))?;
-
-    Ok(Configuration::with_default_ns(server, default_ns))
+#[derive(Debug)]
```

Comment:
probably want to make this `Clone` as well to avoid forcing people to reload it.

Reply:
We can't derive Clone because identity is not clone. We could wrap that in an Arc if we wanted